openmc/capi/mesh.py

```diff
@@ -0,0 +1,173 @@
+from collections.abc import Mapping, Iterable
+from ctypes import c_int, c_int32, c_double, POINTER
+from weakref import WeakValueDictionary
+
+import numpy as np
+from numpy.ctypeslib import as_array
+
+from openmc.exceptions import AllocationError, InvalidIDError
+from . import _dll
+from .core import _FortranObjectWithID
+from .error import _error_handler
+from .material import Material
+
+__all__ = ['Mesh', 'meshes']
+
+# Mesh functions
+_dll.openmc_extend_meshes.argtypes = [c_int32, POINTER(c_int32), POINTER(c_int32)]
+_dll.openmc_extend_meshes.restype = c_int
+_dll.openmc_extend_meshes.errcheck = _error_handler
+_dll.openmc_mesh_get_id.argtypes = [c_int32, POINTER(c_int32)]
+_dll.openmc_mesh_get_id.restype = c_int
+_dll.openmc_mesh_get_id.errcheck = _error_handler
+_dll.openmc_mesh_get_dimension.argtypes = [c_int32, POINTER(POINTER(c_int)), POINTER(c_int)]
+_dll.openmc_mesh_get_dimension.restype = c_int
+_dll.openmc_mesh_get_dimension.errcheck = _error_handler
+_dll.openmc_mesh_get_params.argtypes = [
+    c_int32, POINTER(POINTER(c_double)), POINTER(POINTER(c_double)),
+    POINTER(POINTER(c_double)), POINTER(c_int)]
+_dll.openmc_mesh_get_params.restype = c_int
+_dll.openmc_mesh_get_params.errcheck = _error_handler
+_dll.openmc_mesh_set_id.argtypes = [c_int32, c_int32]
+_dll.openmc_mesh_set_id.restype = c_int
+_dll.openmc_mesh_set_id.errcheck = _error_handler
+_dll.openmc_mesh_set_dimension.argtypes = [c_int32, c_int, POINTER(c_int)]
+_dll.openmc_mesh_set_dimension.restype = c_int
+_dll.openmc_mesh_set_dimension.errcheck = _error_handler
+_dll.openmc_mesh_set_params.argtypes = [
+    c_int32, c_int, POINTER(c_double), POINTER(c_double), POINTER(c_double)]
+_dll.openmc_mesh_set_params.restype = c_int
+_dll.openmc_mesh_set_params.errcheck = _error_handler
+_dll.openmc_get_mesh_index.argtypes = [c_int32, POINTER(c_int32)]
+_dll.openmc_get_mesh_index.restype = c_int
+_dll.openmc_get_mesh_index.errcheck = _error_handler
+
+
+class Mesh(_FortranObjectWithID):
+    """Mesh stored internally.
+
+    This class exposes a mesh that is stored internally in the OpenMC
+    library. To obtain a view of a mesh with a given ID, use the
+    :data:`openmc.capi.meshes` mapping.
+
+    Parameters
+    ----------
+    index : int
+         Index in the `meshes` array.
+
+    Attributes
+    ----------
+    id : int
+        ID of the mesh
+
```

Comment:
would dimension, lower_left, upper_right, and width be considered attributes or do attributes refer to unique identifiers for meshes?

Reply:
Yes, good point. Those are all attributes so I'll add them in to the docstring.